memfs/memory.go

```diff
@@ -11,52 +11,46 @@ import (
 	"time"
 
 	"gopkg.in/src-d/go-billy.v2"
+	"gopkg.in/src-d/go-billy.v2/helper/chroot"
 )
 
 const separator = filepath.Separator
 
 // Memory a very convenient filesystem based on memory files
 type Memory struct {
-	base string
-	s    *storage
+	s *storage
 
 	tempCount int
 }
 
-//New returns a new Memory filesystem
-func New() *Memory {
-	return &Memory{
-		base: string(separator),
-		s:    newStorage(),
-	}
+//New returns a new Memory filesystem.
+func New() billy.Filesystem {
+	fs := &Memory{s: newStorage()}
+	return chroot.New(fs, string(separator))
```

Comment:
Why wrapping it with Chroot?
Maybe it should just implement the `Chroott` function and use `chroot.New` to implement it? Or is this to save the definition of `Chroot` completely?